### ISSUE TITLE
ConfigParser.readfp overspecifies the type of fp argument

### ISSUE THREAD

**Issue by skrap** [python/typeshed#689]
ConfigParser.readfp overspecifies the type of the 'fp' argument.  Despite its name, the documentation for ConfigParser specifically says that it merely needs to be file-like, and only requires a 'readline' method.

Here's what the readfp docs have to say:
>  Read and parse configuration data from the file or file-like object in fp (only the readline() method is used). If filename is omitted and fp has a name attribute, that is used for filename; the default is <???>.

The IO[str] type hint is more than what's actually required in this case, and is causing false positives in my codebase.  I'm not sure what the best solution is, though!

**Comment by ambv** [comment]
That's true. However, what part of the file-like API was required was originally not specified. I did this after the fact to specify the difference between Python 2 and Python 3. In fact, on Python 3, [any iterable is accepted](https://docs.python.org/3/library/configparser.html#configparser.ConfigParser.readfp).

Back to typeshed, it's currently tricky to express structured typing so expressing "anything with a readline() method" is rather hard. Instead, register your type with the TextIOBase ABC (literally `TextIOBase.register(YourClass)`, that should work.

**Comment by JelleZijlstra** [comment]
This could be fixed when we have Protocols, but I don't think there's much we can do before then.
